lib/vintage_net_lte/signal_strength.ex

```diff
@@ -0,0 +1,137 @@
+defmodule VintageNetLTE.SignalStrength do
+  @moduledoc """
+  Functionality for working with the signal strength of an LTE modem
+  """
+
+  @typedoc """
+  Received Signal Strength Indicator (RSSI) is a measurement of how well your
+  device can hear a signal from an access point or router.
+
+  LTE modems use this as the default measurement. See `dbm/1` and `percent/1`
+  to convert into a different measurement.
+  """
+  @type rssi :: non_neg_integer()
+
+  @typedoc """
+  The bit error rate
+
+  Note: for some modems this might always be `99` and can be ignored. For other
+  modems this value is useful. Please see your modem's documentation for a
+  better understanding on how it uses this value.
+  """
+  @type error_rate :: non_neg_integer()
+
+  @type t :: %__MODULE__{
+          rssi: non_neg_integer(),
+          error_rate: non_neg_integer()
+        }
+
+  alias VintageNet.PropertyTable
+
+  defstruct rssi: nil, error_rate: nil
+
+  @doc """
+  Create a new `SignalStrength.t()` data structure.
+
+  Takes an `rssi` and `error_rate`.
+  """
+  @spec new(rssi(), error_rate()) :: t()
+  def new(rssi, error_rate) do
+    %__MODULE__{rssi: rssi, error_rate: error_rate}
+  end
+
+  @doc """
+  Takes a raw AT command from a modem and tries to parse it into a
+  `SignalStrength.t()`
+  """
+  @spec from_at_command(String.t()) :: {:ok, t()} | :error
```

Comment:
This function raises if there's a parse error. For example, if `String.split` doesn't return a 2 element list or the `String.to_integer` calls don't get integers. However, it doesn't raise if the `+CSQ: ` part is messed up. I don't think raising is bad here, but I'd lean to picking one way or the other. Perhaps it would help if AT command parsing was moved out and this module only handled signal strength?